src/services/refreshToken.ts

```diff
@@ -0,0 +1,68 @@
+import axios, { AxiosError } from 'axios';
+
+import { refreshToken as authenticationRefreshToken } from 'adapters/Authentication';
+import { authenticatedHeader } from 'adapters/BaseAuth';
+import { clearToken, getToken, setToken } from 'helpers/authentication';
+import { DeserializableResponse, deserialize } from 'helpers/deserializer';
+import { paths } from 'routes';
+import { Token } from 'types/token';
+
+let isAlreadyFetchingAccessToken = false;
+let subscribers: { (isFetchSuccess: boolean): void }[] = [];
+
+function onAccessTokenFetched(isFetchSuccess: boolean) {
+  subscribers = subscribers.filter((callback) => callback(isFetchSuccess));
+}
+
+function addSubscriber(callback: { (isFetchSuccess: boolean): void }) {
+  subscribers.push(callback);
+}
+
+const doRefreshToken = (token: string) => {
```

Comment:
Should this be `refreshToken` instead?
```suggestion
const doRefreshToken = (refreshToken: string) => {
```

Reply:
Fixed in [de41930](https://github.com/manh-t/react-survey/pull/67/commits/de419303a224598f9b4447ce14e084ddb00e23b9)